src/server/database/Database/Transaction.cpp

```diff
@@ -67,16 +69,26 @@ bool TransactionTask::Execute()
 {
     int errorCode = m_conn->ExecuteTransaction(m_trans);
     if (!errorCode)
+    {
+        m_trans->Cleanup();
         return true;
+    }
 
     if (errorCode == ER_LOCK_DEADLOCK)
     {
         // Make sure only 1 async thread retries a transaction so they don't keep dead-locking each other
-        std::lock_guard<std::mutex> lock(_deadlockLock);
+        //std::lock_guard<std::mutex> lock(_deadlockLock);
         uint8 loopBreaker = 5;  // Handle MySQL Errno 1213 without extending deadlock to the core itself
         for (uint8 i = 0; i < loopBreaker; ++i)
+        {
+            std::this_thread::sleep_for(std::chrono::milliseconds(24 + rand() % 24));
             if (!m_conn->ExecuteTransaction(m_trans))
+            {
+                m_trans->Cleanup();
                 return true;
+            }
+        }
+        ASSERT(false, "Deadlock SQL Transaction");
```

Comment:
shutting down the whole worldserver just for a deadlock sounds too much

Reply:
It shutdowns the server from the assert? How do I make it just a warning?

Reply:
we use TC_LOG_ERROR("category", "message") macro, for example TC_LOG_ERROR("sql.sql", "Deadlock SQL Transaction") .
I'm not sure if current dependency allow to use logs in this file.

Speaking of logs, I would like to log much more info on each retry (how much time it passed since first deadlock, what number of retry are we at, thread ID, etc).

Thanks for the heads-up about the issue in this code, I'll give it some care in the next days when I find time :)